web/models.py

```
@@ -2804,3 +2806,482 @@ class ScheduledPost(models.Model):
+            old_instance = ChapterApplication.objects.get(pk=self.pk)
+            old_status = old_instance.status
+            # If status changed to approved or rejected, set processed_at
+            if self.status in ['approved', 'rejected'] and old_status not in ['approved', 'rejected']:
+                self.processed_at = timezone.now()
+                # Create notification for the user after saving
+                super().save(*args, **kwargs)
+                self._send_status_notification()
+                self._create_membership_if_approved()
+                return
+        # For all other cases, just save normally
+        super().save(*args, **kwargs)
```

Ruff (0.8.2)

Order of model's inner classes, methods, and fields does not follow the Django Style Guide: `save` method should come before custom method

(DJ012)